### ISSUE TITLE
upgrade truffle

### ISSUE THREAD

**Issue by jellegerbrandy** [Paratii-Video/paratii-contracts#51]
We are using truffle 4.1.6

upgradeing to 4.1.7 (which also upgrades solc to 0.4.23) breaks tests in paratii-js

Todo:
- [ ] reproduce the error from apratii-js in a test in paratii-contracts
- [ ] fix and upgrade :-)

https://github.com/trufflesuite/truffle/releases
https://github.com/ethereum/solidity/blob/develop/Changelog.md

**Comment by luisfidelis** [comment]
Hi Jelle. If no one is working on it, i would like to start.

**Comment by jellegerbrandy** [comment]
You are very welcome!

**Comment by jellegerbrandy** [comment]
to reproduce these errors in paratii-js, the road to take is to install paratii-js according to the instructions there, upgrade the reference to paratii-contracts in package.json, run a yarn install, and then run the tests.

**Comment by jellegerbrandy** [comment]
if you have any specific problems, could you signal them here?
https://github.com/Paratii-Video/paratii-portal/issues/656

**Comment by luisfidelis** [comment]
Yes, for sure. I'll try to run ```paratii-js``` tests again and investigate the problem.

**Comment by luisfidelis** [comment]
@jellegerbrandy I've noticed two problems when running ```paratii-js``` tests:
1) I couldn't use parity 1.11.1. After downgrading to 1.9.0 a lot of errors has disappeared.
2) ```ParatiiIPFSLocal``` tests doesn't work well
2.1) ``` it('should allow for file upload', ... )``` expects a different multihash of the file stored.
2.2) ``` it('should add a directory to IPFS',...)``` getting Timeout error.

**Comment by luisfidelis** [comment]
@jellegerbrandy To fix the tests of ```paratii-js``` regarding Solidity 0.4.23/24, i had to add these parameters in the section ```params``` of ```chain.json```:
```
   "eip140Transition": 0,
   "eip211Transition": 0,
   "eip214Transition": 0,
   "eip658Transition": 0
```

It enables byzantium features.
Parity: 1.10.0
